samples/swift/uidemo/ChatViewController.swift

```diff
@@ -0,0 +1,251 @@
+      self.query!.observeEventType(.ChildAdded, withBlock: { [unowned self] _ in
+        self.scrollToBottom(animated: true)
+      })
+    }
+    
+    // Notification boilerplate to handle keyboard appearance/disappearance
+    NSNotificationCenter.defaultCenter().addObserver(self,
+                                                     selector: #selector(keyboardWillShow),
+                                                     name: UIKeyboardWillShowNotification,
+                                                     object: nil)
+    NSNotificationCenter.defaultCenter().addObserver(self,
+                                                     selector: #selector(keyboardWillHide),
+                                                     name: UIKeyboardWillHideNotification,
+                                                     object: nil)
+  }
+  
+  override func viewWillDisappear(animated: Bool) {
+    super.viewWillDisappear(animated)
+    NSNotificationCenter.defaultCenter().removeObserver(self)
+  }
+  
+  deinit {
+    NSNotificationCenter.defaultCenter().removeObserver(self)
+  }
+  
+  @objc private func keyboardWillShow(notification: NSNotification) {
+    let userInfo = notification.userInfo!
+    let endFrameValue = userInfo[UIKeyboardFrameEndUserInfoKey] as! NSValue
+    let endHeight = endFrameValue.CGRectValue().size.height
+    
+    self.bottomConstraint.constant = endHeight
+    
+    let curve = UIViewAnimationCurve(rawValue: userInfo[UIKeyboardAnimationCurveUserInfoKey] as! Int)!
+    let duration = userInfo[UIKeyboardAnimationDurationUserInfoKey] as! Double
+    
+    UIView.setAnimationCurve(curve)
+    UIView.animateWithDuration(duration) { 
+      self.view.layoutIfNeeded()
+    }
+  }
+  
+  @objc private func keyboardWillHide(notification: NSNotification) {
+    self.bottomConstraint.constant = 6
+    
+    let userInfo = notification.userInfo!
+    let curve = UIViewAnimationCurve(rawValue: userInfo[UIKeyboardAnimationCurveUserInfoKey] as! Int)!
+    let duration = userInfo[UIKeyboardAnimationDurationUserInfoKey] as! Double
+    
+    UIView.setAnimationCurve(curve)
+    UIView.animateWithDuration(duration) {
+      self.view.layoutIfNeeded()
+    }
+  }
+  
+  @objc private func didTapSend(sender: AnyObject) {
+    guard let user = self.auth?.currentUser else { return }
+    let uid = user.uid
+    let name = "User " + (uid as NSString).substringToIndex(6)
+    let _text = self.textView.text as String?
+    guard let text = _text else { return }
+    if (text.isEmpty) { return }
+    
+    let chat = Chat(uid: uid, name: name, text: text)
+    
+    self.chatReference.childByAutoId().setValue(chat.dictionary) { (error, dbref) in
+      if let error = error {
+        fatalError("Failed to write message: \(error.localizedDescription)")
+      }
+    }
+    
+    self.textView.text = ""
+  }
+  
+  private func scrollToBottom(animated animated: Bool) {
+    // FirebaseArray's documentation online is pretty lacking.
```

Comment:
Fair play! A reasonable convention in those cases is todo(username):
// TODO(morganchen): Defrobnicate this. 

I'm not totally sold by that type of comment, but at least its clear (for when you come back to look at a bug two years later and realise it has a comment saying "really important to fix this before launch" 🐩 )